lib/scout_apm/logger.ex

```diff
@@ -36,62 +30,24 @@ defmodule ScoutApm.Logger do
   @warn_levels  [:debug, :info, :warn]
   @error_levels [:debug, :info, :warn, :error]
 
+  @log_levels %{
+    debug: @debug_levels,
+    info: @info_levels,
+    warn: @warn_levels,
+    error: @error_levels
+  }
 
-  ################
-  #  Client API  #
-  ################
-
-  def start_link(level \\ @default_level), do: GenServer.start_link(__MODULE__, level, name: @name)
-
-  def set_level(level), do: GenServer.cast(@name, {:set_level, level})
-  def set_enabled(value) when is_boolean(value), do: GenServer.cast(@name, {:set_enabled, value})
-  def enable!(), do: set_enabled(true)
-  def disable!(), do: set_enabled(false)
-
-  def reconfigure_from_config(), do: GenServer.cast(@name, :reconfigure_from_config)
-
-  def debug(msg, metadata \\ []), do: GenServer.cast(@name, {:debug, msg, metadata})
-  def info(msg, metadata \\ []), do: GenServer.cast(@name, {:info, msg, metadata})
-  def warn(msg, metadata \\ []), do: GenServer.cast(@name, {:warn, msg, metadata})
-  def error(msg, metadata \\ []), do: GenServer.cast(@name, {:error, msg, metadata})
-
-  def level(), do: GenServer.call(@name, :level)
-  def enabled?(), do: GenServer.call(@name, :enabled)
-
-  ################
-  #  Server API  #
-  ################
-  def init(level) do
-    state = %{level: level, enabled: true}
-    {:ok, state}
-  end
-
-  def handle_call(:level, _from, %{level: level} = state), do: {:reply, level, state}
-  def handle_call(:enabled, _from, %{enabled: enabled} = state), do: {:reply, enabled, state}
-
-  def handle_cast(:reconfigure_from_config, state) do
-    level = ScoutApm.Config.find(:log_level) || @default_level
+  def log(level, chardata_or_fun, metadata \\ []) when level in @valid_levels do
+    log_level = ScoutApm.Config.find(:log_level) || @default_level
     enabled = ScoutApm.Config.find(:monitor)
-
-    {
-      :noreply,
-      %{state |
-        level: level,
-        enabled: enabled
-      }
-    }
+    key = ScoutApm.Config.find(:key)
+
+    with {:ok, levels} <- enabled && key && Map.fetch(@log_levels, level),
```

Comment:
Can we extract into a `logging_enabled?` function that checks monitor=true, and key is present, rather than the first 2 parts of this conditional.